### PR TITLE
feat(137): add apply AEP

### PR DESCRIPTION
**Pull request by toumorokoshi**
add a new standard method to document apply / PUT.

This AEP is useful for:

- APIs that would like to offer a PUT endpoint, but has lacked guidance from the AEPs.
- APIs that would like to expose a declarative-friendly API  endpoint (the integration of which is simpler than apply).

Apply can also serve as an alternative to expose a create and update endpoint.

This PR also performs a few quick fixes:

- referencing the example proto instead of hard-coding the example.
- some minor fixes on the openapi (removing empty schema)
- minor fixes on the proto (removing invalid fields like 'id')

## 🍱 Types of changes

What types of changes does your code introduce to AEP? _Put an `x` in the boxes
that apply_

- [ ] Enhancement
- [x] [New proposal](https://aep.dev/1#workflow)
- [ ] Migrated from google.aip.dev
- [ ] Chore / Quick Fix

## 📋 Your checklist for this pull request

Please review the [AEP Style and Guidance](https://aep.dev/style-guide) for
contributing to this repository.

### General

- [x] Basic [Guidance](https://aep.dev/style-guide#guidance) is met.
- [x] Ensure that your PR
      [references AEPs](https://aep.dev/style-guide#referencing-aeps)
      correctly.
- [x] [My code has been formatted](https://aep.dev/contributing#formatting)
      (usually `prettier -w .`)

### Additional checklist for a new AEP

- [x] A new AEP **should** be no more than two pages if printed out.
- [x] Ensure that the PR is editable by maintainers.
- [x] Ensure that [File structure](https://aep.dev/style-guide#file-structure)
      guidelines are met.
- [x] Ensure that
      [Document structure](https://aep.dev/style-guide#document-structure)
      guidelines are met.

💝 Thank you!

### CODE REVIEW

**Comment by toumorokoshi** [comment]
This AEP notably does not describe behavior for unpopulated fields - this is waiting on #206 to describe the HTTP behavior, at which point Apply can follow similar guidance.

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:4]:
```suggestion
example, `/v1/publishers/{publisher}/books`) in order to create or replace a new
```

PUT is replace, and we have a standard method called Update with very different semantics, so IMO "update" is confusing here.

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:9]:
```suggestion
replace, and then return, the final resource.
```

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:23]: Maybe we should also answer this for long-running updates...but what happens to the resource is the interim?  Is it write-locked?

Hmm...if I have long-running updates but sync/immediate applies, what happens if I try to apply during a long-running update?

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:72]:
```suggestion
- A `path` field specifying the path of the resource **must** be included.
```

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:78]: This doesn't seem proto-specific - move out of the tab?

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:7]: How will `Resource-oriented design (AEP-121)` get rendered - will it have a lozenge with the AEP's name?  I noticed that below you have a reference to an AEP without a preceding title...

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:139]: Nit - different style than other AEP links directly above.

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [inline aep/general/0137/aep.md.j2:132]: Do we need these?

1.  I thought we were going to make the site generator be smart about rendering AEPs when mentioned by number.
2.  I'm sure I remember us adopting a convention of inline slug-based links, e.g. `[Resource-oriented design](/resources)`.  No?

**Review by rofrankel — CHANGES_REQUESTED** [review]

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:4]: I actually think the semantics will end up pretty similar (with json-merge-patch behavior or replace). But yeah I guess for proto it would be different. SGTM.

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:7]: @earth2marsh suggested the lozenges, and here's an issue to make sure they render similar to the old site: https://github.com/aep-dev/site-generator/issues/52

Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:7]: I'll resolve this one since it's out of scope for the PR, but feel free to re-open if you want me to do something explicit in this one.

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:42]: SGTM - I'll do a follow-up to move this guidance above the tabs (because it's hard to differentiate common guidance and the more specific guidance otherwise).

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:132]: yeah, these are just copy-pasted from other AEPs. But agreed, I'll fix them up,

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:79]: done!

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:23]: It's a good question... 

I don't think we can make "write lock for concurrent updates" a *must* - that's difficult behavior to implement server-side, and we may end up removing a lot of APIs from being AEP-compliant if we have a hard stance on that. 

I do think we have the ability to get some of these patterns though. E.g. for those APIs that do need some level of check-and-set, we do have https://aep.dev/154/. 

I think if we did tackle concurrency, we might be better off filling out https://aep.dev/151/ some more.

WDYT about me starting an issue and we can follow up? aep.dev/134 doesn't have speak to this either.

File: aep/general/0137/aep.md.j2
Comment (gibson042) [inline aep/general/0137/aep.md.j2:14]: This seriously deviates from HTTP, in which PUT against a collection would be interpreted as a request to replace it in entirety (while the request to create a resource within a collection would be POST against the collection or PUT against the URL of the child-to-be [in cases where such a URL is client-provided]). What motivates the divergence?

File: aep/general/0137/aep.md.j2
Comment (gibson042) [inline aep/general/0137/aep.md.j2:54]: This PUT against `…/books/{book}` does not match the described PUT against `…/books`.

**Review by gibson042 — COMMENTED** [review]
What benefit does Apply provide over Create that justifies deviating from specified HTTP behavior?

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:14]: Apologies, I think this was a bad copy-paste when I copied this from create.

The intention is for this to describe a pattern to do a `PUT` against the resource - thereby replacing it. There is no intention to describe a collection-level PUT at the moment. 

I agree with everything you said, I'll fix the typo.

Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2]: the latest commit fixes the typo. Can you review the new wording?

File: aep/general/0137/aep.md.j2
Comment (gibson042) [inline aep/general/0137/aep.md.j2:8]:
```suggestion
method. These operations accept the resource and its path, which it uses to
```

**Review by gibson042 — APPROVED** [review]
I don't think I've seen "Apply" used in this way before, which is why I was originally confused. But it looks good now that the request target has been cleaned up, and fills a significant gap. However, I would recommend adding text to make explicit that APIs may reject attempts to create resources via Apply (e.g. because a collection does not support client-provided paths for its children).

It's also worth referencing this new AEP from 134 (Update).

**Comment by toumorokoshi** [comment]
> I don't think I've seen "Apply" used in this way before, which is why I was originally confused. But it looks good now that the request target has been cleaned up, and fills a significant gap. However, I would recommend adding text to make explicit that APIs may reject attempts to create resources via Apply (e.g. because a collection does not support client-provided paths for its children).
> 
> It's also worth referencing this new AEP from 134 (Update).

Ack! Thanks for the review and approval. Yes, I will add that caveat, and include the reference from AEP 134, before merging.

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [reply aep/general/0137/aep.md.j2:23]: Sure, starting an issue to unblock the PR SGTM.

File: aep/general/0137/aep.md.j2
Comment (rofrankel) [reply aep/general/0137/aep.md.j2]: Is there a reason it needs to be done in a separate PR?

**Review by rofrankel — APPROVED** [review]

File: aep/general/0137/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0137/aep.md.j2:23]: https://github.com/aep-dev/aeps/issues/250

**Comment by toumorokoshi** [comment]
> I don't think I've seen "Apply" used in this way before, which is why I was originally confused. But it looks good now that the request target has been cleaned up, and fills a significant gap. However, I would recommend adding text to make explicit that APIs may reject attempts to create resources via Apply (e.g. because a collection does not support client-provided paths for its children).
> 
> It's also worth referencing this new AEP from 134 (Update).

Also as a quick note - the term "apply" is borrowed from Kubernetes, which uses it to refer to their PUT requests: https://kubernetes.io/docs/reference/kubectl/generated/kubectl_apply/

**Pull request merged**
